src/m3ninx/index/segment/builder/builder.go

```diff
@@ -23,19 +23,38 @@ package builder
 import (
 	"errors"
 	"fmt"
+	"runtime"
+	"sync"
 
 	"github.com/m3db/m3/src/m3ninx/doc"
 	"github.com/m3db/m3/src/m3ninx/index"
 	"github.com/m3db/m3/src/m3ninx/index/segment"
 	"github.com/m3db/m3/src/m3ninx/postings"
 	"github.com/m3db/m3/src/m3ninx/util"
+
+	"github.com/cespare/xxhash"
 )
 
 var (
 	errDocNotFound = errors.New("doc not found")
 )
 
+const (
+	// Slightly buffer the work to avoid blocking main thread.
+	indexQueueSize = 2 << 7
```

Comment:
Nice, yeah perhaps we even increase this to 4096?

Reply:
nit: Also I think I've seen the actual number usually present as a comment, i.e. `indexQueueSize = 2 << 7 // 128`